app/jobs/results_recorder_job.rb

```diff
@@ -1,41 +1,51 @@
+# Preconditions:
+# PlexerJob has made a matching ArchivePreservedCopyPart row
+#
 # Responsibilities:
 # Update DB per event info.
 # Is this event the last needed for the DV to be complete?
 # If NO, do nothing further.
 # If YES, send a message to a non-job pub/sub queue.
 class ResultsRecorderJob < ApplicationJob
   queue_as :endpoint_events
-  attr_accessor :pc, :pcs
+  attr_accessor :apc, :apcs
 
   before_perform do |job|
-    job.pcs ||= PreservedCopy
-                .by_druid(job.arguments.first)
-                .joins(endpoint: [:endpoint_type])
-                .where(
-                  endpoint_types: { endpoint_class: 'archive' },
-                  preserved_copies: { version: job.arguments.second }
-                )
-    job.pc ||= pcs.find_by!('endpoints.delivery_class' => Object.const_get(job.arguments.third))
+    job.apcs ||= ArchivePreservedCopy
+                 .by_druid(job.arguments.first)
+                 .joins(:archive_endpoint)
+                 .where(version: job.arguments.second)
+    job.apc ||= apcs.find_by!('archive_endpoints.delivery_class' => Object.const_get(job.arguments.fourth))
   end
 
   # @param [String] druid
   # @param [Integer] version
+  # @param [String] s3_part_key
   # @param [String] delivery_class Name of the worker class that performed delivery
-  def perform(druid, version, _delivery_class)
-    raise "Status shifted underneath replication: #{pc.inspect}" unless pc.unreplicated?
-    pc.ok!
-    return unless pcs.reload.all?(&:ok?)
+  def perform(druid, version, s3_part_key, _delivery_class)
```

Comment:
though the fourth arg isn't used locally in the `perform` method, it is used by the job (in the `before_perform`), and so i found the leading `_` just a bit confusing (in responding to one of naomi's points below, i was briefly thrown off because i thought this worker did care about delivery class/archive endpoint, and this made me think for a moment that the param was totally disregarded.

anyway, if the convention is that the var gets an underscore if it's not used locally, regardless of whether it's used elsewhere in the job, i'm fine leaving as-is.  i just found it un-intuitive.